bids_conversion/convert_somato_data.py

```diff
@@ -144,30 +141,14 @@
 derivatives_dir = op.join(somato_path_bids, 'derivatives')
 subjects_dir_bids = op.join(derivatives_dir, 'freesurfer', 'subjects')
 
-# Run recon-all from freesurfer
-run_subprocess(['recon-all', '-i', t1w_nii, '-s', '01', '-all'])
+os.environ['SUBJECTS_DIR'] = subjects_dir_bids
 
+run_subprocess(['recon-all', '-i', t1w_nii, '-s', 'sub-01', '-all'])
 # Run make_scalp_surfaces ... use --force to prevent an error from
 # topology defects
-run_subprocess(['mne', 'make_scalp_surfaces', '-s', '01', '--overwrite',
+run_subprocess(['mne', 'make_scalp_surfaces', '-s', 'sub-01', '--overwrite',
```

Comment:
@jasmainak the subject should be `01`, because `sub-` is just the BIDS prefix. The subject label is `01` the same way as a task is not `task-somato` ... but `somato`.

See conversation from here onwards: https://gitter.im/mne-tools/mne-gsoc-2019-BIDS?at=5d31928fa0351b1a8c5dc7af

We are saving the derivative not as `root/derivatives/sub-01/freesurfer ...` but rather acknowledge the freesurfer structure and that subjects are ordered by their label within `root/derivatives/freesurfer/subjects/...`, thus `01` makes more sense and has also been used.

Reply:
hmm ... I am confused. Why wouldn't you want the two subject directories to match? It's okay to have the Freesurfer directory structure. But if you have `sub-01` and `01`, do you have two define two variables in an analysis script?

Reply:
Think of pybids, and the mne-study-template, we always work with the label, not the whole key-value pair:

[` subjects_list = layout.get(return_type='id', target='subject') # will be a list of labels`](https://github.com/mne-tools/mne-study-template/pull/35/files)

so we do:

```Python
sub = '01'
filename = 'sub-{}_task.........'.format(sub)

```

see also my somato PR: https://github.com/mne-tools/mne-python/pull/6414/files

Reply:
okay somato PR convinced me :) Not sure how it would play with pybids but we can see later.